.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,100 @@
+name: Build and test
+
+on:
+  push:
+      branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * 0'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Compile
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: wrapper
+          arguments: compileTestKotlin
+      - name: Build
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build
+      - name: Upload test reports
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: test-reports
+          path: ./build/reports/tests
+      - name: Upload test results
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: test-results
+          path: ./build/test-results
+  build-check:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - run: echo "All build jobs successful."
+  test-integration:
+    runs-on: ubuntu-latest
+    needs: build-check
+    env:
+      JIRA_LICENSE: '${{ secrets.JIRA_LICENSE }}'
+      TERM: dumb
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Prepare Jira license
+        run: echo $JIRA_LICENSE > jira-license.txt
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
```

Comment:
Unfortunately, the creds aren't being refreshed during the long test:
`com.amazonaws.services.s3.model.AmazonS3Exception: The provided token has expired`
Perhaps we can tell it to use STS?

Reply:
Yeah, the default token lifespan is one hour and we can't override this value in the action configuration. This is a limitation of this AWS action and there's even a feature request to allow credential refresh https://github.com/aws-actions/configure-aws-credentials/issues/359.
I tried to extend the session duration on AWS side, lets's see if that works.

Reply:
Ok, I think I solved the issue. I extended the session duration on AWS and in the action configuration to 4 hours and there was no exception related to the expired tokens during the last run.